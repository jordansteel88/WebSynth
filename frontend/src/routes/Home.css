.Home-header {
  margin: -7rem 0 5rem 0;
}
