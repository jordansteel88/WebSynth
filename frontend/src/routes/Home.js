import React from "react";
import "./Home.css";
import Keyboard from '../components/Keyboard';
import EffectsPanel from '../components/EffectsPanel';

const Home = () => {
  return (
    <div className="Home">
      <EffectsPanel /> 
      <Keyboard />
    </div>
  );
}

export default Home;

//currentUser undefined