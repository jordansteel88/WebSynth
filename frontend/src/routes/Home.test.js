import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";
import { BrowserRouter } from "react-router-dom";
import UserContext from '../auth/UserContext';

jest.mock('../components/Synth.js', () => {
  return function DummySynth() {
    return (
      <div>DummySynth</div>
  )};
})

describe("Home", () => {
  const initialState = {
    currentUser: "test"
  }

  it("renders without crashing", function() {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <Home />
        </UserContext.Provider> 
      </BrowserRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ initialState }}>
          <Home />
        </UserContext.Provider> 
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
})