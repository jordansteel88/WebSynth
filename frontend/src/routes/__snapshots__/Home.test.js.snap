// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="EffectsPanel"
    >
      <div>
        DummySynth
      </div>
      <h2>
        Log in or sign up to access more features!
      </h2>
    </div>
    <div
      class="Keyboard"
    >
      <div
        class="Key"
        id="CAPSLOCK"
      >
        <div
          class="Key-text capsLock"
        >
          CAPSLOCK
        </div>
      </div>
      <div
        class="Key-flat"
        id="Q"
      >
        <div
          class="Key-text u-white"
        >
          Q
        </div>
      </div>
      <div
        class="Key"
        id="A"
      >
        <div
          class="Key-text"
        >
          A
        </div>
      </div>
      <div
        class="Key-flat"
        id="W"
      >
        <div
          class="Key-text u-white"
        >
          W
        </div>
      </div>
      <div
        class="Key"
        id="S"
      >
        <div
          class="Key-text"
        >
          S
        </div>
      </div>
      <div
        class="Key-flat"
        id="E"
      >
        <div
          class="Key-text u-white"
        >
          E
        </div>
      </div>
      <div
        class="Key"
        id="D"
      >
        <div
          class="Key-text"
        >
          D
        </div>
      </div>
      <div
        class="Key"
        id="F"
      >
        <div
          class="Key-text"
        >
          F
        </div>
      </div>
      <div
        class="Key-flat"
        id="T"
      >
        <div
          class="Key-text u-white"
        >
          T
        </div>
      </div>
      <div
        class="Key"
        id="G"
      >
        <div
          class="Key-text"
        >
          G
        </div>
      </div>
      <div
        class="Key-flat"
        id="Y"
      >
        <div
          class="Key-text u-white"
        >
          Y
        </div>
      </div>
      <div
        class="Key"
        id="H"
      >
        <div
          class="Key-text"
        >
          H
        </div>
      </div>
      <div
        class="Key"
        id="J"
      >
        <div
          class="Key-text"
        >
          J
        </div>
      </div>
      <div
        class="Key-flat"
        id="I"
      >
        <div
          class="Key-text u-white"
        >
          I
        </div>
      </div>
      <div
        class="Key"
        id="K"
      >
        <div
          class="Key-text"
        >
          K
        </div>
      </div>
      <div
        class="Key-flat"
        id="O"
      >
        <div
          class="Key-text u-white"
        >
          O
        </div>
      </div>
      <div
        class="Key"
        id="L"
      >
        <div
          class="Key-text"
        >
          L
        </div>
      </div>
      <div
        class="Key-flat"
        id="P"
      >
        <div
          class="Key-text u-white"
        >
          P
        </div>
      </div>
      <div
        class="Key"
        id=";"
      >
        <div
          class="Key-text"
        >
          ;
        </div>
      </div>
      <div
        class="Key"
        id="'"
      >
        <div
          class="Key-text"
        >
          '
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
